tests/e2e/tests/e2e_util/util.py

```diff
@@ -225,14 +225,20 @@ def setup_k8s_snap(h: harness.Harness, instance_id: str, snap_path: Path):
 # Validates that the K8s node is in Ready state.
 def wait_until_k8s_ready(h: harness.Harness, instances: List[str]):
     for instance in instances:
-        hostname = (
-            h.exec(instance, ["hostname"], capture_output=True).stdout.decode().strip()
-        )
         result = (
             stubbornly(retries=15, delay_s=5)
             .on(h, instance)
             .until(lambda p: " Ready" in p.stdout.decode())
-            .exec(["k8s", "kubectl", "get", "node", hostname, "--no-headers"])
+            .exec(
+                ["k8s", "kubectl", "get", "node", hostname(h, instance), "--no-headers"]
+            )
```

Comment:
well, that's not really any more readable. 

I like the `util.hostname(...)` method -- that's fine

just keep :

```python
hostname = util.hostname(h, instance)
...
exec(["k8s", "kubectl", "get", "node", hostname, "--no-headers"])
```